server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/DefaultPayloadCodec.java

```diff
@@ -19,18 +19,29 @@
 
 package org.apache.james.blob.objectstorage;
 
+import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 
 import org.jclouds.io.Payload;
 import org.jclouds.io.Payloads;
+import org.jclouds.io.payloads.InputStreamPayload;
 
 public class DefaultPayloadCodec implements PayloadCodec {
     @Override
     public Payload write(InputStream is) {
         return Payloads.newInputStreamPayload(is);
     }
 
+    @Override
+    public Payload write(byte[] bytes) {
```

Comment:
I think that you can check but when you give a ByteArrayInputStream, jcloud sets the content length without

Reply:
There is a byte[] version